cleverhans/attacks.py

```diff
@@ -505,6 +521,20 @@ def parse_params(self,
         return True
 
 
+class BasicIterativeMethod(ProjectedGradientDescent):
+    def __init__(self, model, back='tf', sess=None, dtypestr='float32'):
+        super(BasicIterativeMethod, self).__init__(model, back, sess=sess,
+                                                   dtypestr=dtypestr)
+        self.child = 'BasicIterativeMethod'
```

Comment:
why do we need this self.child variable?

Reply:
my comment above shows how to get rid of this using inheritance / different implementations of the default_rand_minmax method